cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/BasicStreamWriterSizeCollector.java

```diff
@@ -43,6 +67,16 @@ public synchronized void received(String streamName, long dataSize) {
     AtomicLong value = streamSizes.get(streamName);
     if (value == null) {
       value = streamSizes.putIfAbsent(streamName, new AtomicLong(dataSize));
+      if (value != null) {
```

Comment:
`putIfAbsent` returns null if the put succeed, otherwise it returns the existing value. Looks like the condition is inverted here.